### ISSUE TITLE
Date Picker dialogue is opened on first focus of the control, prevents screen reader from reading out the Label attached. Screen reads just reads  "dialogue"

### ISSUE THREAD

**Issue by saidas3524** [OfficeDev/office-ui-fabric-react#8080]
### Environment Information 
 
- __Package version(s)__:  whatever version is used on the portal
- __Browser and OS versions__:  chrome an edge

### Please provide a reproduction of the bug in a codepen:
 1) go to https://codepen.io/saidas3524/pen/daBVPB?&editable=true
2)  use tab to navigate to date picker control. check for the first time focus on datepicker with screen reader
3) check if the screen reader is reading out the label.

 
#### Actual behavior:
 

Label is not read out. Only reader calls as "dialogue"
 
#### Expected behavior:
 Label should be read out. 

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (Yes, No)No 
 
Requested priority: (Blocking, High, Normal, Low) High
 
Products/sites affected: (if applicable)

**Comment by aneeshack4** [comment]
@saidas3524 Which screenreader did you encounter this problem on? Also please specify the version on the portal because I'm not aware which portal this is you're referring to.

**Comment by saidas3524** [comment]
Screen reader used is **NVDA**. Even Narrator can reproduce the same issue. 
**Version** :  6.129.0
**Portal** : https://developer.microsoft.com/en-us/fabric#/components/datepicker

**Comment by aneeshack4** [comment]
@saidas3524 I see that in the codepen, on the first tab the focus goes into calendar view and that starts being announced but when you press Esc, it reads out the label and when you tab to the next element, it reads out the Label of that picker. See the narrator buddy output:
![image](https://user-images.githubusercontent.com/4161791/53369700-a4d95300-3900-11e9-9b83-c2ba2166adc5.png)

I think this might just be because the focus is placed immediately on the date in the open calendar view instead of the picker as a whole.

**Comment by aneeshack4** [comment]
@lorejoh12 Looks like this is actually another bug in the Calendar code... aria-activedescendant is being set on the Calendar table inside the FocusZone of the DatePicker making that the current active child of the DOM and as a result nothing else is being read out first, including the label which is confusing for screenreader users. 
![image](https://user-images.githubusercontent.com/4161791/53373976-e3c0d600-390b-11e9-989b-4cf5b71591da.png)

**Comment by aneeshack4** [comment]
The fix here isn't obvious from the investigation we've done so far - passing this to the UI Fabric Calendar team who owns this component.

**Comment by skunani** [comment]
There are two isses here
1. Onfocus issue : Calendar should not be opened immediately while tabbing through Date
      User Impact: Keyboard only users could feel inconvenience while navigating through the controls if 
      onfocus issue is observed for the calendar
2. JAWS is not reading the labels for the DatePicker control
     User Impact:  Screen Reader users could feel difficulty while working on the application if jaws is not 
      reading the labels for the calendar fields

**Comment by evlevy** [comment]
@skunani I think number 1 (Onfocus issue) is not a problem if you set disableAutoFocus={true} on the DatePicker (although that may cause other focus inconsistencies).

**Comment by lorejoh12** [comment]
Agree with @evlevy, #1 is prevented if you use disableAutoFocus, which was previously requested of us. We did not change the default behavior though to maintain consistency for clients that were already using it as-is that might have their own solutions.

issue 2 is a little more tricky, because what's happening is that the popup is opening before the label has a chance to be read. However, this is also mitigated by disabling auto focus, because that prevents the popout window from opening until the control is invoked, giving JAWS a chance to read the label. I don't know of a way to have JAWS read the label once focus automatically get pushed into the window, so without disabling autofocus I don't think this can be fixed.

Since both of these issues can be solved by adding that property, I'd advocate closing this. One potential breaking change we could make in the future is making disableAutoFocus=true by default (that way consumers will explicitly opt-in to having the broken accessibility). I'll make a note of it when we prepare for 8.0 for date-time.
